src/app/verification-addresses/services/iva.service.ts

```diff
@@ -146,31 +220,53 @@ export class IvaService {
       userId = this.#auth.user()?.id;
       if (!userId) return throwError(() => new Error('Not authenticated'));
     }
-    return this.#http.delete<null>(`${this.#userIvasUrl(userId)}/${ivaId}`);
+    return this.#http
+      .delete<null>(`${this.#userIvasUrl(userId)}/${ivaId}`)
+      .pipe(tap(() => this.#deleteIvaLocally(ivaId)));
```

Comment:
What if there is an error when deleting? Would the pipe not run?

Reply:
Yes, the `tap()` taps only into the happy path. You can also handle the error path in tap(), but handling only the happy case is exactly what we want.